parser.py

```diff
@@ -1 +1,40 @@
-# TODO Xin - Add methods here that we will need. Try to think carefully about organization and how to separate functionality effectively.
\ No newline at end of file
+# TODO Xin - Add methods here that we will need. Try to think carefully about organization and how to separate functionality effectively.
+from bs4 import BeautifulSoup as bs
+from bs4 import SoupStrainer as ss
+import re
+import urllib2 as ul
```

Comment:
As before, "ul" is kind of hard to interpret. If you use urllib2 a LOT, then I could understand shortening the name. But since we barely use it, let's keep the full name.